### PR TITLE
Ensure spread always copies array

### LINKED ISSUE #3096: Copying arrays using spread operator
```js
let a = ['a', 'b', 'c'];
let b = [...a]; // should create an array copy
console.log(a === b); // should be false but it's true when compiled to <ES6
```

### PR DESCRIPTION
**Pull request by ahejlsberg**
Fixes #3096.

### CODE REVIEW

File: tests/baselines/reference/arrayLiteralSpread.js [hunk 1]
```diff
@@ -41,6 +41,6 @@ function f1() {
     var b;
 }
 function f2() {
-    var a = [];
-    var b = [5];
+    var a = [].slice().slice().slice().slice().slice();
+    var b = [5].slice().slice().slice().slice().slice();
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/arrayLiteralSpread.js]: You might not need to slice if you're dealing with an array literal.

Comment (ahejlsberg) [inline tests/baselines/reference/arrayLiteralSpread.js]: The test for which that code is generated is
```typescript
var b = [...[...[...[...[...[5]]]]]];
```
Hardly seems worth optimizing. If you write that, you deserve what we generate. :-)

Comment (ahejlsberg) [inline tests/baselines/reference/arrayLiteralSpread.js]: Actually, the optimization would probably be very cheap. I will take a look.

Comment (ahejlsberg) [inline tests/baselines/reference/arrayLiteralSpread.js]: Ok, done!

**Comment by DanielRosenwasser** [comment]
Looks good! :+1:

**Comment by JsonFreeman** [comment]
:+1:

**Comment by mhegazy** [comment]
merged into release-1.5 under #3209

**Pull request merged**